updates/P7.txt

```diff
@@ -0,0 +1,320 @@
+                           Protocol update
+
+Protocol Version: 7
+Builds on Protocol Version: 6
+
+                               Abstract
+
+This document describes the changes in Concordium protocol version 7 compared to
+protocol version 6.
+
+The protocol change has the following main effects
+- redefines smart contract costs, reducing the cost of smart contract calls;
+- adds support for querying the module reference and contract name of a smart
+  contract instance in smart contracts;
+- changes cooldown behavior so that:
+   * when validators and delegators reduce their stake, the reduction is
+     immediately effective for future stake calculations, and the amount of the
+     reduction is looked for a cooldown period;
+   * validators and delegators can make further changes to their stake while
+     they already have stake in cooldown;
+- disallows shielded (or encrypted) transfer transactions (`TransferToEncrypted`, 
+  `EncryptedAmountTransfer` and `EncryptedAmountTransferWithMemo`).
+
+                              Background
+
+Smart contract execution costs are reduced in protocol version 7 because of
+efficiency improvements in the implementation of smart contract execution.
+The revised cost model is based on an execution model where the interpreter
+stack is compiled to registers.
+
+The new protocol version also imposes a limit on the nesting depth of
+inter-contract calls to 384. This is since each live execution frame can
+require up to 32MB of memory, so limiting the number of active execution
+frames in this way safeguards against excessive memory use.
```

Comment:
I was actually surprised as well for this, I remember discussions about setting 512 as the limit.
Haven't found anything describing why 384 instead.

Reply:
My rationale was that this is safer since with 512 you already reach 16GB which is our stated minimum requirement. 

In reality we could set it to 50 and be fine. 384 has a nice property that it is a bit more than the similar bound on Ethereum :)